src/types.ts

```diff
@@ -86,15 +86,67 @@ export type Blockchain =
       ens?: NamingServiceSource;
       zns?: NamingServiceSource;
       cns?: NamingServiceSource;
-      web3Provider?: Web3Provider;
+      web3Provider?: Provider;
+      provider?: Provider;
     };
 
 export type API = {
   url: string;
 };
 
-export interface Web3Provider {
-  sendAsync: (method: string, params: any) => Promise<any>;
+type ProviderParams = readonly [
+  (
+    | {
+        [record: string]: string;
+      }
+    | string
+  ),
+  string,
+];
+
+export interface RequestArguments {
+  method: string;
+  params?: ProviderParams;
+}
+
+export interface Provider {
+  request: (request: RequestArguments) => Promise<any>;
+}
+
+/**
+ * @deprecated
+ */
+export type Web3Provider = Provider;
```

Comment:
This isn't a correct alias any longer. If we are breaking the old functionality (that wasn't working anyway), probably it's just better to get rid of it. What do you think @bogdan?